raster/src/main/scala/geotrellis/raster/DelayedConversionMultibandTile.scala

```diff
@@ -0,0 +1,313 @@
+package geotrellis.raster
+
+import geotrellis.raster.resample._
+import geotrellis.vector.Extent
+
+import spire.syntax.cfor._
+
+/**
+  * [DelayedConversionTile]] represents a tile that wraps an inner tile,
+  * and for any operation that returns a Tile, returns an ArrayTile with
+  * a cell type of the target cell type.
+  */
+class DelayedConversionMultibandTile(inner: MultibandTile, override val targetCellType: CellType)
+  extends MultibandTile with MacroMultibandCombiners {
+
+  private def validateBand(i: Int) = assert(i < bandCount, s"Band index out of bounds. Band Count: $bandCount Requested Band Index: $i")
+
+  val cols = inner.cols
+  val rows = inner.rows
+
+  def cellType: CellType =
+    inner.cellType
+
+  def convert(cellType: CellType): MultibandTile =
+    inner.convert(cellType)
+
+  def withNoData(noDataValue: Option[Double]): MultibandTile =
+    inner.withNoData(noDataValue)
+
+  def interpretAs(newCellType: CellType): MultibandTile =
+    withNoData(None).convert(newCellType)
+
+  def bandCount: Int = inner.bandCount
+  def band(bandIndex: Int): Tile = inner.band(bandIndex)
+  def bands: Vector[Tile] = inner.bands
+
+  def subsetBands(bandSequence: Seq[Int]): MultibandTile =
+    inner.subsetBands(bandSequence)
+
+  def foreach(f: (Int, Int) => Unit): Unit = inner.foreach(f)
+  def foreachDouble(f: (Int, Double) => Unit): Unit = inner.foreachDouble(f)
+  def foreach(b0: Int)(f: Int => Unit): Unit = inner.foreach(b0)(f)
+  def foreachDouble(b0: Int)(f: Double => Unit): Unit = inner.foreachDouble(b0)(f)
+  def foreach(f: Array[Int] => Unit): Unit = inner.foreach(f)
+  def foreachDouble(f: Array[Double] => Unit): Unit = inner.foreachDouble(f)
+
+  /**
+    * Map over a subset of the bands of an [[ArrayMultibandTile]] to
+    * create a new integer-valued [[MultibandTile]] that have the target [[CellType]].
+    *
+    * @param   subset  A sequence containing the subset of bands that are of interest.
+    * @param   f       A function to map over the bands.
+    * @return          A MultibandTile containing the result
+    */
+  def map(subset: Seq[Int])(f: (Int, Int) => Int): MultibandTile = {
+    val newBands = Array.ofDim[Tile](bandCount)
+    val set = subset.toSet
+
+    subset.foreach({ b =>
+      require(0 <= b && b < bandCount, "All elements of subset must be present")
+    })
+
+    (0 until bandCount).foreach({ b =>
+      val bandTile = band(b).delayedConversion(targetCellType)
+
+      if (set.contains(b))
+        newBands(b) = bandTile.map({ z => f(b, z) })
+      else if (targetCellType.isFloatingPoint)
+        newBands(b) = bandTile.mapDouble({ z => z })
+      else
```

Comment:
They need to change cell type, so I changed these to use `convert` to be more explicit